packages/react-router/tests/loaders.test.tsx

```diff
@@ -729,3 +730,89 @@ test('clears pendingTimeout when match resolves', async () => {
   expect(nestedPendingComponentOnMountMock).not.toHaveBeenCalled()
   expect(fooPendingComponentOnMountMock).not.toHaveBeenCalled()
 })
+
+test('reproducer for #4696', async () => {
+  const rootRoute = createRootRoute({
+    beforeLoad: async ({ context }) => {
+      return {
+        ...context,
+        isAuthenticated: true,
+        isAdmin: false,
+      }
+    },
+    loader: ({ context }) => context,
+    pendingComponent: () => 'Loading...',
+    wrapInSuspense: true,
+    component: RootRouteContent,
+  })
+
+  function RootRouteContent() {
+    const routeData = rootRoute.useLoaderData()
+    const isAuthenticated = routeData.isAuthenticated
+    return (
+      <>
+        {!!isAuthenticated && (
+          <nav style={{ display: 'flex', gap: 8 }}>
+            <Link to="/">Index</Link>
+            <Link to="/dashboard">Dashboard</Link>
+          </nav>
+        )}
+        <hr />
+        <Outlet />
+      </>
+    )
+  }
+
+  const indexRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/',
+    component: () => <h1>Index</h1>,
+  })
+
+  const dashboardRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/dashboard',
+    pendingComponent: () => 'Loading dashboard...',
+    validateSearch: () => {
+      return {
+        page: 0,
+      }
+    },
+    component: () => <h1>Dashboard</h1>,
+  })
+
+  const routeTree = rootRoute.addChildren([indexRoute, dashboardRoute])
+  const router = createRouter({ routeTree })
+
+  render(<RouterProvider router={router} />)
+  await act(async () => {
+    /* no-op */
+  })
+  expect(screen.getByRole('heading', { name: 'Index' })).toBeInTheDocument()
```

Comment:
please use getByTestId and use `data-testid="..."`
we try not to use getByRole in new tests anymore

Reply:
got it – why's that? (just curious)

Reply:
it allows to change text, tags etc without having to update the getByRole calls
you just need to make sure the test-id is correctly specified